### PR TITLE
Lrp for review

### PR DESCRIPTION
**Pull request by Nimanui**
Looking to add some alpha, beta, and epsilon LRP saliency options run across a few different models (both the MIMIC IV stagenet and the chest x-ray CNN)
Original implementation is based off of  https://arxiv.org/abs/1604.00825 and is very vibey so I'm expecting some comments
I may prune down the example options a bit as this ended up a bit bloated, and I'll need to add some more documentation

### CODE REVIEW

File: pyhealth/interpret/methods/basic_gradient.py
Comment (Nimanui) [inline pyhealth/interpret/methods/basic_gradient.py:1]: This is just some refactoring to pull out the visualization components for use with other saliency methods, I don't know how we may want to formalize visualization going forward or otherwise evolve that.

File: pyhealth/interpret/methods/lrp.py [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+            elif not isinstance(target_class_idx, torch.Tensor):
+                target_class_idx = torch.tensor(
+                    target_class_idx, device=logits.device
+                )
+            batch_size = logits.size(0)
+            output_relevance = torch.zeros_like(logits)
+            output_relevance[range(batch_size), target_class_idx] = logits[
+                range(batch_size), target_class_idx
+            ]
+            return output_relevance
```
Comment (Logiquo) [inline pyhealth/interpret/methods/lrp.py:431]: The target_class_index hanlding has been unified in https://github.com/sunlabuiuc/PyHealth/pull/926. With the following semantics
1. binary will always have targret_class_index 0
2. the attribution is always towards positive, and up to the evaluator to flip the sign for negative class contribution
3. if not specified, both multi-class and multi-label uses argmax

File: pyhealth/interpret/methods/lrp.py [hunk 2]
```diff
@@ -0,0 +1,778 @@
...
+    LinearLRPHandler,
+    ReLULRPHandler,
+    Conv2dLRPHandler,
+    MaxPool2dLRPHandler,
+    AvgPool2dLRPHandler,
+    AdaptiveAvgPool2dLRPHandler,
+    BatchNorm2dLRPHandler,
+    FlattenLRPHandler,
+    DropoutLRPHandler,
+    RNNLRPHandler,
```
Comment (Logiquo) [inline pyhealth/interpret/methods/lrp.py:34]: Do we need special handling for Conv1d, Sigmoid, Tanh, Softmax in StageNet?

File: pyhealth/interpret/methods/lrp.py [hunk 3]
```diff
@@ -0,0 +1,778 @@
...
+                return bname
+        return None
+
+    def _propagate_through_residual_block(
+        self,
+        block_name: str,
+        block_layer_names_rev: list,
+        r_in: torch.Tensor,
+        ctx: _LRPHookContext,
+    ) -> torch.Tensor:
```
Comment (Logiquo) [inline pyhealth/interpret/methods/lrp.py:452]: this seems to ignore `rule` args for LRP and hard code to epsilon rule regardless if it is epsilon rule or alpha beta?

File: pyhealth/interpret/methods/lrp_base.py [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+
+        s = relevance_output / z_total
+        out_pad = conv_output_padding(layer, z_pos.shape, x.shape)
+        trans_kw = dict(
+            stride=layer.stride, padding=layer.padding,
+            output_padding=out_pad, dilation=layer.dilation, groups=layer.groups,
+        )
+        c_pos = crop_spatial(F.conv_transpose2d(s, W_pos, None, **trans_kw), x.shape)
+        c_neg = crop_spatial(F.conv_transpose2d(s, W_neg, None, **trans_kw), x.shape)
+        return x * (alpha * c_pos + beta * c_neg)
```
Comment (Logiquo) [inline pyhealth/interpret/methods/lrp_base.py:396]: The linear uses x_p * (alpha * c_pos - beta * c_neg). Is it expected for them to be different?

**Review by Logiquo — CHANGES_REQUESTED** [review]
